File: 17s2/test07/distinct_lines.pl
#!/usr/bin/perl -w

if (@ARGV != 1) {
	die;
}
$case = 0;
$max_distinct_lines = $ARGV[0];
@tem = ();
$line_cnt = 0;
while ( $line = <STDIN> ) {
	$line_cnt++;
	$line = uc $line;
	$line =~ s/\n//;
	$line =~ s/\ +/\ /g;
    $line =~ s/^\ +//;
    #print "line = $line\n";
	if (grep {$_ eq $line} @tem) {
		next;
	}
	push @tem, $line;
    #print "@tem\n";
	#print "$#tem\n";
	if ($#tem eq $max_distinct_lines-1) {
		print "$max_distinct_lines distinct lines seen after $line_cnt lines read.\n";
        $case = 1;
	}
}
if ($case eq 0) {
    print "End of input reached after $line_cnt lines read - $max_distinct_lines different lines not seen.\n";
}

File: 17s2/test09/distinct_lines.py
#!/usr/bin/python3

import sys,re

end_tag = int(sys.argv[1])
final_data = []
line_cnt = 0

def deal_line(lines):
        lines = re.sub(r'(\s)+',' ',lines)
        lines = re.sub(r'^(\s)*','',lines)
        lines = re.sub(r'(\s)*$','',lines)
        lines = lines.strip()
        d = lines.split(r"$")
        element = d[0]
        doc = element.lower()
        return doc

try:
    while True:
        lin = input()
        line_cnt+=1
        x = deal_line(lin)
        #print(lines)
        
        
        #print("---",doc)
        
        if x not in final_data:
            final_data.append(x)
        #print(concise_data)
        
        length = len(final_data)
        if end_tag == length:
            print("%d distinct lines seen after %d lines read." % (end_tag,line_cnt))
            sys.exit()

except EOFError:
    print("End of input reached after %d lines read -  %d different lines not seen." % (line_cnt,end_tag))
    sys.exit()
